### PR TITLE
Feature/add instance segmentation

### PR DESCRIPTION
**Pull request by Fireronin**
Added support for classes and entities,
Things to do in future:
Resolve 1. todo done
Move constant of 20 COMPRESSED_ID_BIT_DEPTH to some other place done
Initialize tags from engine and not from csv  -- done

### CODE REVIEW

File: Code/Source/Entity/EntityManager.cpp
Comment (PawelLiberadzki) [inline Code/Source/Entity/EntityManager.cpp:123]: Line 104 above - I feel like this case here and the one above both should return default id, that is either 0 or 1 or whatever we will agree one. Just be sure it is consistent.

Comment (Fireronin) [reply Code/Source/Entity/EntityManager.cpp:123]: yaaa, my bad in general that thing on line 104 was still in class 0, because 1 is packed value, i changed this a bit and now it should make more sense and be more self consistent

File: Code/Source/Utilities/RGLUtils.cpp
Comment (PawelLiberadzki) [reply Code/Source/Utilities/RGLUtils.cpp]: What's with this one?

File: Code/Source/Entity/EntityManager.h
Comment (PawelLiberadzki) [inline Code/Source/Entity/EntityManager.h:59]: Nit: one line spacing will be enough.

File: Code/Source/Entity/EntityManager.cpp
Comment (PawelLiberadzki) [inline Code/Source/Entity/EntityManager.cpp:125]: New line here.

File: Code/Source/Entity/EntityManager.cpp
Comment (PawelLiberadzki) [reply Code/Source/Entity/EntityManager.cpp]: What did you decided to do with this?

File: Code/Source/Entity/EntityManager.cpp
Comment (PawelLiberadzki) [reply Code/Source/Entity/EntityManager.cpp]: Updated version is nice!

File: Code/CMakeLists.txt
Comment (PawelLiberadzki) [inline Code/CMakeLists.txt:99]: Please review also other builds - there should be Dockers for it.

File: Code/Source/Entity/EntityManager.cpp
Comment (Fireronin) [reply Code/Source/Entity/EntityManager.cpp]: Generally, I couldn't get this to compile with the other function, and in theory either way I would need to do same loop, so might as well do this like this

**Comment by pijaro** [comment]
@Fireronin Please, update the README.md as well.

File: README.md
Comment (PawelLiberadzki) [inline README.md:121]: Is this maybe possible to force from UI? Or maybe it is already being done like that?

File: static/png/segmentation_classes.png
Comment (PawelLiberadzki) [inline static/png/segmentation_classes.png:1]: This image has to be updated @Fireronin, please. We agreed on specific names for these parameters. This is because user will not know what should be passed to parameter called "unsigned char".

Comment (Fireronin) [reply static/png/segmentation_classes.png:1]: Oh, this might be due to this being auto generated from tuple

File: README.md
Comment (Fireronin) [reply README.md:121]: We can leave this as an issue, @michalpelka agreed that it's ok to expect that user can read

File: README.md
Comment (Fireronin) [reply README.md:117]: That's true

File: static/png/segmentation_classes.png
Comment (Fireronin) [reply static/png/segmentation_classes.png:1]: I'm opening an issue for this as it's beyond the scope of this pr.
https://github.com/RobotecAI/o3de-rgl-gem/issues/28

File: Code/Source/Entity/EntityManager.h
Comment (michalpelka) [inline Code/Source/Entity/EntityManager.h:41]: Consider adding an alias for this reused type :
AZStd::shared_ptr<AZStd::set<AZStd::pair<AZStd::string, uint8_t>>> and 
AZStd::set<AZStd::pair<AZStd::string, uint8_t>>.
I understand that AZStd::set (instead of AZStd::map) is needed due to reflection in o3de?

File: Code/Source/Entity/EntityManager.h
Comment (michalpelka) [inline Code/Source/Entity/EntityManager.h:52]: consider:
```suggestion
        int32_t GetRglId() const;
```

File: Code/Source/Entity/MeshEntityManager.h
Comment (michalpelka) [inline Code/Source/Entity/MeshEntityManager.h:37]: Why this line was removed? It is correct to have base class hint here

File: Code/Source/Entity/EntityManager.h
Comment (michalpelka) [inline Code/Source/Entity/EntityManager.h:77]: Consider
```suggestion
        AZStd::shared_ptr<AZStd::unorder_set<AZStd::pair<AZStd::string, uint8_t>>> m_classNameToClassIdMapping;
```

**Review by michalpelka — COMMENTED** [review]
More or less nitpicking

File: Code/Source/Entity/EntityManager.h
Comment (Fireronin) [reply Code/Source/Entity/EntityManager.h:41]: As a fun fact in o3de this is kept as a vector as vector has better gui generation

File: Code/Source/Entity/MeshEntityManager.h
Comment (Fireronin) [reply Code/Source/Entity/MeshEntityManager.h:37]: By mistake

**Comment by Fireronin** [comment]
I changed holding of class tags to unordered set as requested

**Review by PawelLiberadzki — APPROVED** [review]
Good job!

**Review by PawelLiberadzki — CHANGES_REQUESTED** [review]
Just in case of sudden merge, I block this merge to development here, because it can break other projects.

File: Code/Source/Utilities/RGLUtils.cpp
Comment (alek-kam-robotec-ai) [inline Code/Source/Utilities/RGLUtils.cpp:53]: Simply use `RGL_CHECK` instead as it is already defined in this way.

File: Code/Source/Utilities/RGLUtils.cpp
Comment (alek-kam-robotec-ai) [inline Code/Source/Utilities/RGLUtils.cpp:61]: Same as before.

File: Code/Source/Entity/MeshEntityManager.cpp
Comment (alek-kam-robotec-ai) [inline Code/Source/Entity/MeshEntityManager.cpp:19]: Nit: Use angle brackets in the include directives. This is partly to ensure using include paths (quotes allow for relative pathes which are prohbited by the [coding standard](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#include-paths)) and partly to maintain a unified convention throughout the project. Please apply this to all newly added the include directives.

File: Code/Source/Entity/EntityManager.h
Comment (alek-kam-robotec-ai) [inline Code/Source/Entity/EntityManager.h:42]: Why not `const Utils::ClassTags&`?

File: Code/Source/Utilities/RGLUtils.h
Comment (alek-kam-robotec-ai) [inline Code/Source/Utilities/RGLUtils.h:18]: Why not `using ClassTags = ...`?

**Comment by alek-kam-robotec-ai** [comment]
Before merging, please check this pr's mention in #37.

**Comment by alek-kam-robotec-ai** [comment]
I'm closing this pr in favour of #49.

**Comment by alek-kam-robotec-ai** [comment]
I'm closing this pr in favour of #49.

**Pull request closed without merging**
